cranelift/codegen/src/machinst/abi.rs

```diff
@@ -2069,7 +2072,7 @@ impl<M: ABIMachineSpec> Callee<M> {
 #[derive(Clone, Debug)]
 pub enum ArgLoc {
     Reg(PReg),
-    Stack(StackAMode),
+    Stack { offset: i64, ty: ir::Type },
```

Comment:
Can we document what base this is an offset from? SP right?

Reply:
(And presumably this change is because we don't need the full generality of `StackAMode` anymore?)

Reply:
I guess the offset is either SP or FP depending on whether the call is a tail call or not? That is pretty subtle and seems relatively easy to mix up. Is there a reason we can't keep using `StackAMode` here? That would avoid mix ups because we would always have to match on the amode to determine whether we are dealing with SP or FP offsets. I think it would be fine to have `unreachable!()` arms in these matches if we always use SP for regular calls and FP for tail calls, but avoiding the mix-up footgun is nice.

Reply:
This offset always describes the number of bytes into the argument area. Whether we find the argument area relative to FP or SP depends on how `gen_arg`'s caller is set up, not on ABI details. That's why we switched away from encoding that decision here.